src/main/java/org/opensearch/ad/transport/GetAnomalyDetectorTransportAction.java

```diff
@@ -1,430 +1,438 @@
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+import com.google.common.collect.Sets;
+
+public class GetAnomalyDetectorTransportAction {
+    // extends HandledTransportAction<GetAnomalyDetectorRequest, GetAnomalyDetectorResponse> {
+
+    private static final Logger LOG = LogManager.getLogger(GetAnomalyDetectorTransportAction.class);
+
+    private final SDKClusterService clusterService;
+    private final SDKRestClient client;
+
+    private final Set<String> allProfileTypeStrs;
+    private final Set<DetectorProfileName> allProfileTypes;
+    private final Set<DetectorProfileName> defaultDetectorProfileTypes;
+    private final Set<String> allEntityProfileTypeStrs;
+    private final Set<EntityProfileName> allEntityProfileTypes;
+    private final Set<EntityProfileName> defaultEntityProfileTypes;
+    private final NamedXContentRegistry xContentRegistry;
+    private final DiscoveryNodeFilterer nodeFilter;
+    private final TransportService transportService;
+    private volatile Boolean filterByEnabled;
+    private final ADTaskManager adTaskManager;
+
+    @Inject
+    public GetAnomalyDetectorTransportAction(
+        TransportService transportService,
+        DiscoveryNodeFilterer nodeFilter,
+        ActionFilters actionFilters,
+        SDKClusterService clusterService,
+        SDKRestClient client,
+        Settings settings,
+        NamedXContentRegistry xContentRegistry,
+        ADTaskManager adTaskManager
+    ) {
+        // super(GetAnomalyDetectorAction.NAME, transportService, actionFilters, GetAnomalyDetectorRequest::new);
+        this.clusterService = clusterService;
+        this.client = client;
+
+        List<DetectorProfileName> allProfiles = Arrays.asList(DetectorProfileName.values());
+        this.allProfileTypes = EnumSet.copyOf(allProfiles);
+        this.allProfileTypeStrs = getProfileListStrs(allProfiles);
+        List<DetectorProfileName> defaultProfiles = Arrays.asList(DetectorProfileName.ERROR, DetectorProfileName.STATE);
+        this.defaultDetectorProfileTypes = new HashSet<DetectorProfileName>(defaultProfiles);
+
+        List<EntityProfileName> allEntityProfiles = Arrays.asList(EntityProfileName.values());
+        this.allEntityProfileTypes = EnumSet.copyOf(allEntityProfiles);
+        this.allEntityProfileTypeStrs = getProfileListStrs(allEntityProfiles);
+        List<EntityProfileName> defaultEntityProfiles = Arrays.asList(EntityProfileName.STATE);
+        this.defaultEntityProfileTypes = new HashSet<EntityProfileName>(defaultEntityProfiles);
+
+        this.xContentRegistry = xContentRegistry;
+        this.nodeFilter = nodeFilter;
+        filterByEnabled = AnomalyDetectorSettings.FILTER_BY_BACKEND_ROLES.get(settings);
+        try {
+            clusterService.getClusterSettings().addSettingsUpdateConsumer(FILTER_BY_BACKEND_ROLES, it -> filterByEnabled = it);
+        } catch (Exception e) {
+            // FIXME Handle this
```

Comment:
https://github.com/opensearch-project/opensearch-sdk-java/issues/422 has been closed, no need to surround the `addSettingsUpdateConsumer `with a try/catch

Reply:
Thanks for pointing to the issue. I went ahead and removed the FIXME (try/catch) for all the other files as well.